app/Http/Controllers/Api/Client/Servers/LogsController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Pterodactyl\Http\Controllers\Api\Client\Servers;
+
+use Illuminate\Http\Request;
+use Pterodactyl\Models\Server;
+use Pterodactyl\Models\AuditLog;
+use Pterodactyl\Models\Permission;
+use Illuminate\Auth\Access\AuthorizationException;
+use Pterodactyl\Transformers\Api\Client\AuditLogTransformer;
+use Pterodactyl\Http\Controllers\Api\Client\ClientApiController;
+
+class LogsController extends ClientApiController
+{
+    /**
+     * LogsController constructor.
+     */
+    public function __construct() {
+        parent::__construct();
+    }
+
+    /**
+     * Returns all the logs for a given server instance in a paginated
+     * result set.
+     *
+     * @throws \Illuminate\Auth\Access\AuthorizationException
+     */
+    public function index(Request $request, Server $server): array
```

Comment:
You should be using a ClientApiRequest class instead of manually verifying permissions. It'll check permissions for you, all you have to do is use the new request type here instead of Illuminate Request

Look at any other client controller for examples

Requests folder: https://github.com/pterodactyl/panel/tree/develop/app/Http/Requests/Api/Client/Servers